docs/plugins/customization.md

```diff
@@ -0,0 +1,69 @@
+---
+id: customization
+title: Customization
+description: Documentation on adding a customization logic to the plugin
+---
+
+## Overview
+
+The Backstage core logic provides a possibility to make the component customizable in such a way that the application
+developer can redefine the labels, icons, elements or even completely replace the component. It's up to each plugin
+to decide what can be customized.
+
+## For a plugin developer
+
+When you are creating your plugin, you have a possibility to use a metadata field and define there all
+customizable elements. For example
+
+```typescript jsx
+export type MyPluginMetadataProps = {
+  createButtonTitle: string;
+};
+
+export const myPluginMetadataRef = createMetadataRef<MyPluginMetadataProps>({
+  id: 'MyPluginMetadataProvider',
+});
+
+export const myPluginMetadata = {
+  MyPluginMetadataProvider: {
+    // or [myPluginMetadataRef.id]
+    createButtonTitle: 'Create Component',
+  },
+};
+
+const plugin = createPlugin({
+  id: 'myPlugin',
+  metadata: [myPluginMetadata],
+});
+```
+
+And the rendering part of the exposed component can retrieve that metadata as:
+
+```typescript jsx
+export function DefaultMyPluginWelcomePage(props: DefaultMyPluginWelcomeProps) {
+  const { createButtonTitle } =
+    useMetadata<DefaultMyPluginWelcomeProps>(myPluginMetadataRef);
```

Comment:
@Rugvip 
Would `useMetadata<DefaultMyPluginWelcomeProps>(key, props)` be an option for that? When props are provided as an argument for useMetadata, they will be merged. 

As not all components might be using useMetadata, and other way around, not all components have props.

Reply:
No I think that's a bit of a different use-case tbh. I'm thinking more plugin-wide pre-processing of the configuration so that each component doesn't need to do the work locally

Reply:
Sorry, i'm not really sure what you mean.

`useMetadata` returns back the merge of default data with reconfiguration, and plugin developer can use it wherever he/she wants it. Inside the component or plugin wise. 


Like for example if I'm developing a component and have a hierarchy of components inside like 
```
 <PageWithHeader title={`${orgName} Catalog`} themeId="home">
      <EntityListProvider>
        <Content>
         ....
          <CatalogFilterLayout>
            <CatalogFilterLayout.Filters>
              <EntityTagPicker />
```

And my intention here to make customisable `EntityTagPicker`, I'd prefer to call `useMetadata` inside `EntityTagPicker`, then passing props to all components. 

And it's only my preference, this implementation doesn't restrict to it, plugin developer still can do it.

Can I see it is more like changing the example in the documentation?

Or if some code changes have to be done, can you please give some small pseudo-code, to make it easier to understand your thought?